src/server.c

```diff
@@ -5275,8 +5254,8 @@ const char *replstateToString(int replstate) {
     case SLAVE_STATE_WAIT_BGSAVE_START:
     case SLAVE_STATE_WAIT_BGSAVE_END:
         return "wait_bgsave";
-    case SLAVE_STATE_BACKGROUND_RDB_LOAD:
-        return "psync";
+    case SLAVE_STATE_ACTIVE_RDB_CHAN:
+        return "bg_transfer";
```

Comment:
> let's be sure to document all the interface changes in the top comment.

Do you mean one general comment which explains the flow and states of rdb-channel session (like Replication buffer blocks)? Or specifically this new state?

Reply:
i mean a bullet list of all the interface changes (configs, commands, command arguments and command replies).
in this case:
* the `state` sub-field in the `slave%d` info field (`INFO REPLICATION`), has a new `bg_transfer` state. (in addition to `wait_bgsave`, `send_bulk`, `online`)

Reply:
Before resolving this comment, does [Fix comments and add high level design top comment and diagram](https://github.com/redis/redis/pull/12109/commits/79eae09b1f76fd4567b870e424b9c450392a90f3) enough? Or should I go into more details like you mentioned?

Reply:
not sure how this is related.
i meant the top comment of the PR (which will later become the commit comment when squash-merging).

i.e. that comment should describe the motivation, benefits, design, interface changes, any regressions or additional unrelated changes, etc. so that someone can read that comment and know all the important bits without reading the code